File: clients/node/examples/user-flow.js
import { Screen, Pattern } from "../src";

async function main() {
  const screen = await Screen({
    startupTimeoutMs: 10_000
  });
  try {
    const pattern = Pattern("assets/pattern.png").exact();
    const match = await screen.click(pattern);
    console.log(`clicked match target at (${match.targetX}, ${match.targetY})`);
    console.log("automation actions sent");
  } finally {
    await screen.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: packages/api/cmd/sikuli-go/examples/python/user-flow.py
from __future__ import annotations

from bootstrap import ensure_sikuli_go_on_path
from sikuligo import Pattern, Screen

ensure_sikuli_go_on_path()

screen = Screen(startup_timeout_seconds=10.0)
try:
    match = screen.click(Pattern("assets/pattern.png").exact())
    print(f"clicked match target at ({match.target_x}, {match.target_y})")
    print("automation actions sent")
finally:
    screen.close()
